public/app/features/explore/Table/TableContainer.tsx

```diff
@@ -51,11 +52,12 @@ export class TableContainer extends PureComponent<Props> {
   }
 
   getTableTitle(dataFrames: DataFrame[] | null, data: DataFrame, i: number) {
-    let title = data.name ? `Table - ${data.name}` : 'Table';
-    if (dataFrames && dataFrames.length > 1) {
-      title = `Table - ${data.name || data.refId || i}`;
+    let name: string | number | undefined = data.name;
+    if (!name && (dataFrames?.length ?? 0) > 1) {
+      name = data.refId || i;
```

Comment:
Just nitpicking :D something like below would probably be more readable to me:

```suggestion
    let name = data.name;
    if (!name && (dataFrames?.length ?? 0) > 1) {
      name = data.refId || `${i}`;
```